controller/grpc/controller.proto

```diff
@@ -0,0 +1,275 @@
+}
+
+message Port {
+  int32 port = 1;
+  string proto = 2;
+  HostService service = 3;
+}
+
+message VolumeReq {
+  string path = 1;
+  bool delete_on_stop = 2;
+}
+
+// See github.com/flynn/flynn/host/types Mount
+message HostService {
+  string display_name = 1;
+  // Create the service in service discovery
+  bool create = 2;
+  HostHealthCheck check = 3;
+}
+
+// See github.com/flynn/flynn/host/types Mount
+message HostHealthCheck {
+  // Type is one of tcp, http, https
+  string type = 1;
+  // Interval is the time to wait between checks after the service has been
+  // marked as up. It defaults to two seconds.
+  google.protobuf.Duration interval = 3;
+  // Threshold is the number of consecutive checks of the same status before
+  // a service will be marked as up or down after coming up for the first
+  // time. It defaults to 2.
+  int32 threshold = 4;
+  // If KillDown is true, the job will be killed if the service goes down (or
+  // does not come up)
+  bool kill_down = 5;
+  // StartTimeout is the maximum duration that a service can take to come up
+  // for the first time if KillDown is true. It defaults to ten seconds.
+  google.protobuf.Duration start_timeout = 6;
+
+  // Extra optional config fields for http/https checks
+  string path = 7;
+  string host = 8;
+  string match = 9;
+  int32 status = 10;
+}
+
+// See github.com/flynn/flynn/host/resource Spec
+message HostResourceSpec {
+  // Request, if set, is the amount of resource a job expects to consume,
+  // so the job should only be placed on a host with at least this amount
+  // of resource available, and once scheduled this amount of resource
+  // should then be unavailable on the given host.
+  int64 request = 1;
+  // Limit, if set, is an upper limit on the amount of resource a job can
+  // consume, the outcome of hitting this limit being implementation
+  // defined (e.g. a system error, throttling, catchable / uncatchable
+  // signals etc.)
+  int64 limit = 2;
+}
+
+// See github.com/flynn/flynn/host/types Mount
+message HostMount {
+  string location = 1;
+  string target = 2;
+  bool writable = 3;
+  string device = 4;
+  string data = 5;
+  int32 flags = 6;
+}
+
+// See github.com/opencontainers/runc/libcontainer/configs Device
+message LibContainerDevice {
+  // Device type, block, char, etc.
+  int32 type = 1;
+  // Path to the device.
+  string path = 2;
+  // Major is the device's major number.
+  int64 major = 3;
+  // Minor is the device's minor number.
+  int64 minor = 4;
+  // Cgroup permissions format, rwm.
+  string permissions = 5;
+  // FileMode permission bits for the device.
+  uint32 file_mode = 6;
+  // Uid of the device.
+  uint32 uid = 7;
+  // Gid of the device.
+  uint32 gid = 8;
+  // Write the file to the allowed list
+  bool allow = 9;
+}
+
+// See github.com/flynn/flynn/logaggregator/client Message
+message LogChunk {
+  // host_id is the host that the job was running on when this log message was
+  // emitted.
+  string host_id = 1;
```

Comment:
Should this be a reference to a host resource? (e.g. `hosts/HOST_ID`)

Reply:
Makes sense.